### ISSUE TITLE
Specifying a value for security_group_ids ignores any values assigned for vpc_id and subnet_id

### ISSUE THREAD

**Issue by mariusmitrofan** [hashicorp/packer#12370]
#### Overview of the Issue

I understand that you guys would like everyone to be able to use the product even the unexperienced ones.

Hence the logic of setting custom value for "associate_public_ip_address" depending on default/non-default VPC for instance.

HOWEVER, this hinders the development process for more experienced users.

One such cases is that when you specify a value for "security_group_ids", any and whatsoever value you specify via "vpc_id", "subnet_id", "subnet_filter", "vpc_filter" gets totally ignored and the following happens:
- the VPC is autodiscovered via the security group definition
- a random subnet is assigned
- a random availability zone is assigned

This is infuriating for me as I have to use specific subnets within that VPC in order to establish connection to specific services.

#### Reproduction Steps

Create an "amazon-ebs" source with the following definition:

```
  subnet_id                   = "subnet-asdashfjdsakjd"
  associate_public_ip_address = false
  vpc_id = "vpc-asdjsakdkaskad"
  security_group_ids          = [var.security_group_id]
```

### Packer version

1.8.2

### Simplified Packer Template

Not relevant

### Operating system and Environment details

Not relevant

### Log Fragments and crash.log files

Not relevant

**Comment by mariusmitrofan** [comment]
"Good" news ->

If you don't specify the value for "associate_public_ip_address", it works as intended.

Aka this works ->

```
  security_group_ids          = [var.security_group_id]
  source_ami                  = data.amazon-ami.this.id
  subnet_id                   = "subnet-asdsadsa"
```

But this doesn't ->

```
  security_group_ids          = [var.security_group_id]
  source_ami                  = data.amazon-ami.this.id
  subnet_id                   = "subnet-asdsadsa"
  associate_public_ip_address = false
```

**Comment by nywilken** [comment]
Hi @mariusmitrofan thanks for bubbling up this issue. We fixed a problem with the `associate_public_ip_address` with v1.2.3 of the plugin (See https://github.com/hashicorp/packer-plugin-amazon/issues/18), but in doing so, we forgot to handle one case, leading to this problem, apologies for that. Refer to https://github.com/hashicorp/packer-plugin-amazon/pull/370 for details. 

The fix to the Amazon plugin has been released in [v1.2.4](https://github.com/hashicorp/packer-plugin-amazon/releases/tag/v1.2.4). Please upgrade to the latest plugin version and let us know if you are still experiencing the problem.

Apologies for the unexpected breakage
